concepts/Enarx_Public_Chat_App_Zulip/000005-Enarx_Public_Chat_App_Zulip.md

```diff
@@ -0,0 +1,126 @@
+
+In terms of implementation, it seems fairly straightforward, and there are examples of this being done on an organizational level (see [here](https://leanprover-community.github.io/archive/)). An Enarx deployment of this tool would look fairly similar. We would need to create a new repository within the Enarx organization to hold the chat record, though it shouldn't need to be maintained after `zulip-archive` is set up.
+
+## Rationale and alternatives
+
+We think Zulip is the superior choice due to a number of feature advantages it has over Gitter and other solutions.
+- Organization
+    - Zulip has topic-based organization, allowing better separation of thoughts and tasks than Gitter, which only allows for a chronological feed.
+    - Like Gitter, it supports multiple rooms (known as “streams”). It also promotes stream discoverability better than Gitter.
+    - Zulip supports chat search, where Gitter offers nothing of the sort.
+- App support (especially mobile)
+    - Gitter’s mobile app is, to put it kindly, not very good. It doesn’t support notifications at all, it is slow, and it doesn’t register read messages well. Additionally, Gitter themselves do not officially recommend their own apps, and [say](https://gitter.im/apps) they may be deprecated in the future. Zulip’s mobile apps work well, do support notifications, and generally support a larger subset of features.
+    - Both have desktop Linux apps available, though both are also available in browser.
+    - Both have Windows and Mac desktop clients.
+- Nice-to-have features
+    - Zulip supports file uploads, where Gitter does not. This can be helpful if we need to share a screenshot, for example.
+    - Zulip allows users to link to a unique message in the chat.
+    - Zulip appears to have much more robust integration support, including Github integration. Gitter’s Github integration is out of date and does not work.
+
+In terms of alternatives: Following Enarx's design principles of open development, we strongly value FOSS solutions, of which there are, unfortunately, few. Beyond Zulip, there's:
+
+- Gitter, which we've already discussed
+- Matrix, which has some nice security/privacy features, but otherwise looks high-maintenance and not necessarily as feature-complete as Zulip.
+- IRC, which is a non-starter.
+
+## Prior art
+
+We have used Gitter for a while, and have had annoyances with it for a while, as previously discussed. Beyond that, Enarx has not used any other public chat apps.
+
+## Unresolved questions
+
+We still have no reference point for how reliable Zulip is. This, unfortunately, is something we can only learn with experience.
+
+## Future Possibilities
+
+One particularly exciting possibility is Zulip's robust integration potential. Zulip has a powerful [bot](https://zulipchat.com/api/running-bots) framework, which could potentially open up interesting automation opportunities with Github and elsewhere.
```

Comment:
This is both true and already frustrating. I expect Zulip's integration with other non-chat platforms is better than it is with chat platforms, where, as mentioned in the RFC document, you can only sync a Zulip **topic**  to another platform (which has very limited usefulness, given many topics will be created). But i expect the integration with other platforms that are complementary will be better.

Reply:
@axelsimon I'm not sure what else could be done. If you support topics (Zulip) and you want to sync to a platform that doesn't support topics (IRC), the best possible integration is to lock IRC to a single topic.

Reply:
That's a good point Nathaniel. I think i'd just at least sync everything, adding the topic to the message text, rather than only sync one topic, but at least a switch to chose which of these two workarounds to use would be nice. I suspect locking to a single topic can also have its use in that you can have a topic for, say, monitoring, and have that replicated in other places.

Anyway, that's taking us on a bit of a tangent.